### PR TITLE
daemon: Daemon.newContainer: inline Daemon.generateHostname

### PR DESCRIPTION
**Pull request by thaJeztah**
generateHostname was only used here, and defined as a method on Daemon without depending on the daemon type. Inline the function and simplify the logic (as both the "network=host" and non-host code only had to be executed if no hostname was set).

### CODE REVIEW

**Review by austinvazquez — APPROVED** [review]

File: daemon/container.go [hunk 1]
```diff
@@ -154,7 +155,7 @@ func (daemon *Daemon) newContainer(name string, operatingSystem string, config *
 	base.Name = name
 	base.Driver = daemon.imageService.StorageDriver()
 	base.OS = operatingSystem
-	return base, err
+	return base, nil
```
Comment (AkihiroSuda) [inline daemon/container.go:158]: Probably this can be a separate commit

**Review by AkihiroSuda — APPROVED** [review]

File: daemon/container.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/container.go:158]: Oh! I wanted to, but looks like I didn't do so. Let me put it separate.

Comment (thaJeztah) [reply daemon/container.go:175]: done 👍

**Review by AkihiroSuda — APPROVED** [review]
